pkg/apps/application-list.jsx

```diff
@@ -0,0 +1,168 @@
+    }
+
+    render() {
+        var self = this;
+        var comp = self.props.comp;
+        var state = self.state;
+
+        function navigate() {
+            cockpit.location.go(comp.id);
+        }
+
+        function action(func, arg, progress_title) {
+            self.setState({ progress_title: progress_title });
+            func(arg, (data) => self.setState({ progress: data })).
+                always(() => self.setState({ progress: false })).
+                fail(show_error);
+        }
+
+        function install() {
+            action(PackageKit.install, comp.pkgname, _("Installing"));
+        }
+
+        function remove() {
+            action(PackageKit.remove, comp.file, _("Removing"));
+        }
+
+        var name, summary_or_progress, button;
+
+        if (comp.installed) {
+            name = <a onClick={left_click(() => launch(comp))}>{comp.name}</a>;
+        } else {
+            name = comp.name;
+        }
+
+        if (state.progress) {
+            summary_or_progress = <ProgressBar title={state.progress_title} data={state.progress}/>;
+            button = <CancelButton data={state.progress}/>;
+        } else {
+            if (state.error) {
+                summary_or_progress = (
+                    <div>
+                        {comp.summary}
+                        <div className="alert alert-danger alert-dismissable">
+                            <button className="close" aria-hidden="true"
+                                    onClick={left_click(() => { this.setState({ error: null }) })}>
+                                <span className="pficon pficon-close"/>
+                            </button>
+                            <span className="pficon pficon-error-circle-o"/>
+                            {state.error}
+                        </div>
+                    </div>
+                );
+            } else {
+                summary_or_progress = comp.summary;
+            }
+
+            if (comp.installed) {
+                button = <button className="btn btn-danger" onClick={left_click(remove)}>{_("Remove")}</button>;
+            } else {
+                button = <button className="btn btn-default" onClick={left_click(install)}>{_("Install")}</button>;
+            }
+        }
+
+        return (
+            <tr onClick={left_click(navigate)}>
+                <td><img src={icon_url(comp.icon)}/></td>
+                <td>{name}</td>
+                <td>{summary_or_progress}</td>
+                <td>{button}</td>
+            </tr>
+        );
+    }
+}
+
+class ApplicationList extends React.Component {
+    constructor() {
+        super();
+        this.state = { progress: false };
+    }
+
+    render() {
+        var self = this;
+        var comps = [ ];
+        for (var id in this.props.metainfo_db.components)
+            comps.push(this.props.metainfo_db.components[id]);
+        comps.sort(function (a, b) { return a.name.localeCompare(b.name); });
```

Comment:
Style nitpick: Since you use `=>` everywhere else, this could also be written with that. Gets rid of the `return`, `function`, and the semicolon :)

Reply:
`comps.sort((a, b) => a.name.localeCompare(b.name));`

Reply:
Done.